src/arch/linux/async/signal.c

```diff
@@ -492,7 +492,11 @@ signal_pre_init(void)
 #else
   newsetsig(SIGILL, abort_signal);
   newsetsig(SIGTRAP, abort_signal);
+#ifdef __APPLE__
+  newsetsig(SIGBUS, minfault);
```

Comment:
But doesn't SIGBUS mean that
backing-store is unmapped?
You can get SIGBUS on linux as
well, if you don't ftruncate() your shm.

Reply:
i think it does SIGBUS on any shm, but let me get a small test case just in case

Reply:
this program, which calls `ftruncate`, produces SIGSEGV on Linux but SIGBUS on Mac:

```
#include <unistd.h>
#include <stdio.h>
#include <sys/mman.h>
#include <fcntl.h>

int main(void)
{
     int fd = shm_open("/dosemu0000", O_RDWR | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
     shm_unlink("/dosemu0000");
     if (fd == -1) {
        perror("shm_open");
        return 0;
     }
     ftruncate(fd, 4096);
     char *a = mmap(NULL, 4096, PROT_READ, MAP_SHARED, fd, 0);
     if (a == MAP_FAILED) {
       perror("mmap");
       return 0;
     }
     a[0] = 22;
     return 0;
}
```